src/terrain/routes/search.clj

```diff
@@ -15,12 +15,20 @@
   (util/optional-routes
     [config/search-routes-enabled]
 
-    (GET "/filesystem/search-documentation" [:as req]
-      (util/controller req c-search/get-data-search-documentation))
-    (POST "/filesystem/search" [:as req]
-      (util/controller req c-search/do-data-search :params :body))
+    (context "/filesystem" []
+       :tags ["filesystem"]
 
-    (GET "/filesystem/index" [q tags & opts]
-      (if (or q tags)
-        (search/search (:shortUsername user/current-user) q tags opts)
-        (missing-arg-response "`q` or `tags`")))))
+       (GET "/search-documentation" []
+            :summary "Get additional documentation for the search endpoint"
+            :description "Returns documentation of the available querydsl clauses and their
+            arguments/types, plus the list of available sort fields."
+            (ok (c-search/get-data-search-documentation)))
+
+       (POST "/search" []
+             :summary "Perform a data search"
+             :body [body (doc-only SearchQuery SearchQueryDocs)]
+             :description "Search utilizing the querydsl.
+             This endpoint automatically filters results to those the user can see, and adds a
+             \"permission\" field that summarizes the requesting user's effective permission on each result."
```

Comment:
These strings will be formatted like Markdown, so you can also use back-ticks to format `permission` as code.